### PR TITLE
Align all usages of Jackson to be 2.14.0

### PR DESCRIPTION
**Pull request by jakelandis**
related: #90553

---
Note - I do not plan to back port this to 7.x since #90553 was not back ported and am not confident such a large upgrade across the board for 7.x is wise.

### CODE REVIEW

**Comment by jakelandis** [comment]
@ChrisHegarty can you take a quick look for anything I may have missed specific to 2.14.0 ?
@rjernst - IIRC there is some history around difficulty upgrading this dependency, can you take a quick look to see if I missed any gotcha's ?

**Comment by jakelandis** [comment]
@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by jakelandis** [comment]
I don't get what is going on with CI ... the first run was success, the second run had a few failures due dependency verification issues ( due to no longer listing multiple elder versions hashes), the third run had a single failure. None of the failures will reproduce locally.  

@elasticsearchmachine run elasticsearch-ci/part-1

**Review by rjernst — COMMENTED** [review]
In general I think we should be moving away from this global version declaration style. It makes upgrading very difficult, mostly because of third party libs that often have complicated reasons they can't be upgraded. We should upgrade simple uses aggressively (eg x-content, where it matters most for security), and 3rd party libs as necessary/possible.

File: libs/x-content/impl/build.gradle [hunk 1]
```diff
@@ -10,15 +10,13 @@ apply plugin: 'elasticsearch.java'
 
 archivesBaseName = "x-content-impl"
 
-String jacksonVersion = "2.14.0"
```
[the remark this answers is not in the document]
Comment (ChrisHegarty) [reply libs/x-content/impl/build.gradle:13]: Agreed - my preference is to retain this x-content specific version. The ability to upgrade the version of Jackson independent of other parts of the system is a feature here (not a bug).

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
I will close this PR and replace it with an upgrade to 2.14.2 
https://github.com/elastic/elasticsearch/pull/93438
the suggestions regarding x-content's jackson version and repository-azure plugin jackson version are applied

**Pull request closed without merging**